src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -498,11 +499,16 @@ static void xocl_work_cb(struct work_struct *work)
 		if (!xocl_get_buddy_fpga(&buddy_xdev, xocl_get_buddy_cb))
 			buddy_xdev = NULL;
 		if (buddy_xdev)
-			(void) xocl_hot_reset(buddy_xdev, XOCL_RESET_FORCE |
+			ret = xocl_hot_reset(buddy_xdev, XOCL_RESET_FORCE |
 				XOCL_RESET_SHUTDOWN | XOCL_RESET_NO);
-		(void) xocl_hot_reset(xdev, XOCL_RESET_FORCE |
+		ret = xocl_hot_reset(xdev, XOCL_RESET_FORCE |
 			       	XOCL_RESET_SHUTDOWN);
 		mutex_unlock(&xocl_reset_mutex);
+		if (ret) {
+			xocl_xdev_err(xdev, "reset failed with error %d", ret);
+			xocl_reset_notify(xdev->core.pdev, false);
+			xocl_drvinst_set_offline(xdev->core.drm, false);
+		}
 		break;
```

Comment:
if you handle the failure case in xocl_hot_reset(), no need to check the return code here. otherwise, you may need to check the return code of xocl_hot_reset() at other spots also
It'd better to introduce another sysfs node to save the reset status, and tell the user whether the reset succeeds or not with this sysfs node.